drivers/sensor/tdk/icm456xx/icm456xx.c

```diff
@@ -0,0 +1,1061 @@
+{
+	int64_t conv_val;
+
+	/* 16 bit accelerometer. 2^15 bits represent the range in G */
+	/* see datasheet section 3.2 for details */
+	conv_val = (int64_t)raw_val * SENSOR_G * fs / INT16_MAX;
+
+	val->val1 = conv_val / 1000000;
+	val->val2 = conv_val % 1000000;
+}
+
+void icm456xx_convert_gyro(struct sensor_value *val, int16_t raw_val, uint16_t fs)
+{
+        int64_t conv_val;
+
+        /* 16 bit gyroscope. 2^15 bits represent the range in degrees/s */
+        /* see datasheet section 3.1 for details */
+        conv_val = ((int64_t)raw_val * fs * SENSOR_PI) / (INT16_MAX * 180U);
+
+        val->val1 = conv_val / 1000000;
+        val->val2 = conv_val % 1000000;
+}
+
+static void icm456xx_convert_temp(struct sensor_value *val, int16_t raw_val)
+{
+	int64_t conv_val;
+
+	/* see datasheet section 15.9 for details */
+	conv_val = 25 * 1000000 + ((int64_t)raw_val * 1000000 / 2);
+	val->val1 = conv_val / 1000000;
+	val->val2 = conv_val % 1000000;
+}
+
+static int icm456xx_channel_get(const struct device *dev, enum sensor_channel chan,
+				struct sensor_value *val)
+{
+	int res = 0;
+	struct icm456xx_data *data = dev->data;
+#ifdef CONFIG_TDK_APEX
+	const struct icm456xx_config *cfg = dev->config;
+#endif
+
+	icm456xx_lock(dev);
+	if (chan == SENSOR_CHAN_ACCEL_XYZ) {
+		icm456xx_convert_accel(&val[0], data->accel_x, data->accel_fs);
+		icm456xx_convert_accel(&val[1], data->accel_y, data->accel_fs);
+		icm456xx_convert_accel(&val[2], data->accel_z, data->accel_fs);
+	} else if (chan == SENSOR_CHAN_ACCEL_X) {
+		icm456xx_convert_accel(val, data->accel_x, data->accel_fs);
+	} else if (chan == SENSOR_CHAN_ACCEL_Y) {
+		icm456xx_convert_accel(val, data->accel_y, data->accel_fs);
+	} else if (chan == SENSOR_CHAN_ACCEL_Z) {
+		icm456xx_convert_accel(val, data->accel_z, data->accel_fs);
+	} else if (chan == SENSOR_CHAN_GYRO_XYZ) {
+		icm456xx_convert_gyro(&val[0], data->gyro_x, data->gyro_fs);
+		icm456xx_convert_gyro(&val[1], data->gyro_y, data->gyro_fs);
+		icm456xx_convert_gyro(&val[2], data->gyro_z, data->gyro_fs);
+	} else if (chan == SENSOR_CHAN_GYRO_X) {
+		icm456xx_convert_gyro(val, data->gyro_x, data->gyro_fs);
+	} else if (chan == SENSOR_CHAN_GYRO_Y) {
+		icm456xx_convert_gyro(val, data->gyro_y, data->gyro_fs);
+	} else if (chan == SENSOR_CHAN_GYRO_Z) {
+		icm456xx_convert_gyro(val, data->gyro_z, data->gyro_fs);
+	} else if (chan == SENSOR_CHAN_DIE_TEMP) {
+		icm456xx_convert_temp(val, data->temp);
+	} else {
+		res = -ENOTSUP;
+	}
+
+
+	icm456xx_unlock(dev);
+
+	return res;
+}
+
+#ifdef CONFIG_ICM456XX_TRIGGER
+static int icm456xx_fetch_from_fifo(const struct device *dev)
+{
+	struct icm456xx_data *data = dev->data;
+	int status = 0;
+	inv_imu_int_state_t int_state;
+	inv_imu_sensor_data_t sd;
+	inv_imu_fifo_data_t fd;
+
+	status |= inv_imu_get_int_status(&data->driver, INV_IMU_INT1, &int_state);
+	if (int_state.INV_UI_DRDY) {
+		status |= inv_imu_get_register_data(&data->driver, &sd);
```

Comment:
Here you get data from registers instead of reading the FIFO as defined in the API name.

Reply:
Removed